### PR TITLE
More docs in reference manual and add architecture doc

### PR DESCRIPTION
**Pull request by dai-chen**
*Issue #, if available:* https://github.com/opendistro-for-elasticsearch/sql/issues/380

*Description of changes:*

1. Add more docs in reference manual: https://github.com/dai-chen/sql/blob/automated-docs-phase-3/docs/user/index.rst
2. Add architecture doc written by Peng: https://github.com/dai-chen/sql/blob/automated-docs-phase-3/docs/dev/Architecture.md

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: docs/dev/Architecture.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+2. The Syntax and Semantic Analyzer will walk through the AST and verify whether the query is follow the grammar and supported by the OD-SQL. e.g. *SELECT * FROM semantics WHERE LOG(age, city) = 1, *will throw exception with message* Function [LOG] cannot work with [INTEGER, KEYWORD]. *and sample usage message* Usage: LOG(NUMBER T) → DOUBLE.
+3. The Druid Lex&Parser takes the input query and generate the druid AST which is different from the AST generated by the ANTRL. This module is the open source library (https://github.com/alibaba/druid) used by NLPChina originally.
+4. The QueryPlanner Builder take the AST as input and generate the LogicalPlan from it. Then it optimize the LogicalPlan to PhysicalPlan.(In current implementation, only rule-based model is implemented). The major part of PhysicalPlan generation use NLPChina’s original logic to translate the SQL expression in AST to Elasticsearch DSL.
+5. The QueryPlanner executor execute the PhysicalPlan in worker thread.
+6. The formatter will reformatted the response data to the required format. The default format is JDBC format.
```
Comment (abbashus) [inline docs/dev/Architecture.md:38]: np: `reformatted ` -> `reformat `

Comment (dai-chen) [reply docs/dev/Architecture.md:38]: Will do. Thanks!

**Comment by dai-chen** [comment]
> LGTM, thanks!
> NP for this pr: could you correct the typo: [make](https://github.com/dai-chen/sql/blame/automated-docs-phase-3/docs/developing.rst#L61) -> mkdir

Sure, will fix it. Thanks!

File: docs/dev/Architecture.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
+# OpenDistro SQL Engine Architecture
+
+---
+## 1.Overview
+
+The OpenDistro SQL (OD-SQL) project is developed based on NLPChina project (https://github.com/NLPchina/elasticsearch-sql) which has been deprecated now ([attributions](https://github.com/opendistro-for-elasticsearch/sql/blob/master/docs/attributions.md)). Over the one year in development, a lot of features have been added to the OD-SQL project on top of the existing older nlp-china project. The purpose of this document is to explain the OD-SQL current architecture going ahead.
```
Comment (davidcui-amzn) [inline docs/dev/Architecture.md:6]: np: should we maintain consistency with how we refer to NLPChina? e.g NLPChina vs nlp-china

Comment (dai-chen) [reply docs/dev/Architecture.md:6]: Good catch. Will stick with NLPChina. Thanks!

**Pull request merged**